src/meshes/p4est_mesh.jl

```diff
@@ -48,7 +55,16 @@ mutable struct P4estMesh{NDIMS, RealT <: Real, IsParallel, P, Ghost, NDIMSP2, NN
         ghost = ghost_new_p4est(p4est)
         ghost_pw = PointerWrapper(ghost)
 
-        mesh = new{NDIMS, eltype(tree_node_coordinates), typeof(is_parallel),
+        # To enable the treatment of a manifold of dimension NDIM embedded within an
+        # ambient space of dimension NDIMS_AMBIENT, we store both as type parameters and
+        # allow them to differ in the general case. This functionality is used in the
+        # (currently alpha-stage) package TrixiAtmo.jl for constructing discretizations
+        # on spherical shell domains for applications in global atmospheric modelling.
+        # The ambient dimension NDIMS_AMBIENT is therefore set here in the inner
+        # constructor to size(tree_node_coordinates, 1).
+        # See https://github.com/trixi-framework/Trixi.jl/pull/2068.
```

Comment:
I would opt for something more persistent, i.e., maybe drop the reference to TixiAtmo (or just mention it without any referring to state of the project). 
I am also not sure if the link to the PR is really necessary here, I do not think that it is good practice to introduce them as comments in the code.